src/classes/storage.ts

```diff
@@ -33,4 +64,70 @@ export class Storage {
    * @returns {void}
    */
   public deleteStopCluster(): void {}
+
+  /**
+   * @description Adding StopCluster to the ground places list.
+   * @param {GroundPlaceGenerated} stopCluster - The stop cluster to add to the ground places list.
+   * @returns {void}
+   */
+  public addStopClusterToGroundPlacesList({
+    id: stopClusterGpuid,
+    name,
+    latitude,
+    longitude,
+    countryCode,
+    type,
+  }: GroundPlaceGenerated): void {
+    // Create the StopCluster
+    const newStopCluster = {
+      [stopClusterGpuid]: { name, latitude, longitude, country_code: countryCode, type, childs: [] },
+    };
+
+    // TODO: Call checkValidity on the groundPlacesList to check if all rules are respected before adding the StopCluster
+
+    // Append the StopCluster to the ground places list.
+    this.groundPlacesList = {
+      ...this.groundPlacesList,
+      ...newStopCluster,
+    };
+  }
+
+  /**
+   * @description Adding StopGroup to the ground places list and also referenced it inside it's StopCluster parent.
+   * @param {StopClusterGpuid} stopClusterParentGpuid - The StopCluster parent ground place unique identifier.
+   * @param {GroundPlaceGenerated} stopGroup - The StopGroup to add to the ground places list.
+   */
+  public addStopGroupToGroundPlacesList(
+    stopClusterParentGpuid: StopClusterGpuid,
+    { id: stopGroupGpuid, name, latitude, longitude, countryCode, type }: GroundPlaceGenerated,
+  ): void {
+    // Create the StopGroup
+    const newStopGroup = {
+      [stopGroupGpuid]: {
+        country_code: countryCode,
+        name,
+        longitude,
+        latitude,
+        type,
+        childs: [],
+      },
+    };
+
+    // TODO: Call checkValidity on the groundPlacesList to check if all rules are respected before adding the StopCluster
+
+    // Add the StopGroup Gpuid inside it's StopCluster parent
+    const stopClusterParent = {
+      [stopClusterParentGpuid]: {
+        ...this.groundPlacesList[stopClusterParentGpuid],
+        childs: [...this.groundPlacesList[stopClusterParentGpuid].childs, stopGroupGpuid],
+      } as StopCluster, // Type casting here otherwise TS will complain since groundPlacesList could have two structure of places..
+    };
+
+    // Append to the ground places list.
+    this.groundPlacesList = {
```

Comment:
Again. Super strange stuff

Reply:
Removed, better using separate methods like `create`, `remove`, `move` that are driven by the `groundPlacesDiff` object